### PR TITLE
TASK-40251: update  notification message when inviting user to a space 

### PR DESCRIPTION
**Pull request by nidhalnidhal**
I modified the invitation message in the green box  by adding the invited user and also updated the notification message by adding the space manager

### CODE REVIEW

File: component/notification/src/main/java/org/exoplatform/social/notification/channel/template/WebTemplateProvider.java [hunk 1]
```diff
@@ -53,6 +53,8 @@
 import org.exoplatform.social.notification.plugin.*;
 import org.exoplatform.webui.utils.TimeConvertUtils;
 
+import javax.ws.rs.core.SecurityContext;
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/channel/template/WebTemplateProvider.java]: unneeded import

File: component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java [hunk 2]
```diff
@@ -29,13 +29,16 @@
 import org.exoplatform.commons.utils.CommonsUtils;
 import org.exoplatform.services.log.ExoLogger;
 import org.exoplatform.services.log.Log;
+import org.exoplatform.services.rest.tools.DummySecurityContext;
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java]: unneeded import

File: component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java [hunk 3]
```diff
@@ -29,13 +29,16 @@
...
 import org.exoplatform.social.core.space.SpaceListenerPlugin;
 import org.exoplatform.social.core.space.model.Space;
 import org.exoplatform.social.core.space.spi.SpaceLifeCycleEvent;
 import org.exoplatform.social.notification.plugin.RequestJoinSpacePlugin;
 import org.exoplatform.social.notification.plugin.SocialNotificationUtils;
 import org.exoplatform.social.notification.plugin.SpaceInvitationPlugin;
 
+import javax.ws.rs.core.SecurityContext;
+import java.security.Principal;
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java]: unneeded import

File: component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java [hunk 4]
```diff
@@ -114,9 +117,10 @@ public void spaceAccessEdited(SpaceLifeCycleEvent event) {}
   public void addInvitedUser(SpaceLifeCycleEvent event) {
     Space space = event.getSpace();
     String userId = event.getTarget();
-    
+    String SenderName = space.getEditor();
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java]: Any manager could invite the user, I think the sender should be the one who did really send the invitation

File: component/notification/src/main/java/org/exoplatform/social/notification/plugin/SpaceInvitationPlugin.java [hunk 5]
```diff
@@ -22,6 +22,8 @@
 import org.exoplatform.container.xml.InitParams;
 import org.exoplatform.social.core.space.model.Space;
 
+
+
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/plugin/SpaceInvitationPlugin.java]: To remove, please always keep just the needed changes

File: extension/notification/src/main/webapp/WEB-INF/notification/templates/SpaceInvitationPlugin.gtmpl [hunk 6]
```diff
@@ -32,10 +32,12 @@
...
+                                                     String sender = "<a target=\"_blank\" style=\"color: #2f5e92; text-decoration: none; font-weight: bold; font-size:13px; font-family: HelveticaNeue,Helvetica,Arial,sans-serif;\" href=\"#\">" + _ctx.escapeHTML(SENDER_NAME) + "</a>";
+                                                     String spaceUrl = "<a target=\"_blank\" style=\"color: #2f5e92; text-decoration: none; font-weight: bold; font-size:13px; font-family: HelveticaNeue,Helvetica,Arial,sans-serif;\" href=\""+ SPACE_URL + "\">" + _ctx.escapeHTML(SPACE) + "</a>";
                                                     %>
-                                                    <%=_ctx.appRes("Notification.message.SpaceInvitationPlugin", spaceUrl)%>
-                                                </p>
+                                                    <%=_ctx.appRes("Notification.intranet.message.SpaceInvitationPlugin",sender)%>
+                                                   <div class="sendertest"><%= SENDER_NAME %></div>
```
Comment (ahamdi) [inline extension/notification/src/main/webapp/WEB-INF/notification/templates/SpaceInvitationPlugin.gtmpl]: Is this CSS class needed ?

File: webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue [hunk 7]
```diff
@@ -63,7 +65,7 @@
           <v-list-item-content>
             <v-list-item-title class="externalUserEmail" v-text="user"></v-list-item-title>
             <v-list-item-subtitle class="subEmail">{{ $t('peopleList.label.pending') }}</v-list-item-subtitle>
-          </v-list-item-content>
+          </v-list-item-content>g
```
Comment (ahamdi) [inline webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue]: a type to remove

File: webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue [hunk 8]
```diff
@@ -161,6 +163,16 @@ export default {
...
+    },
+
+    invitedUser() {
+      if ( this.invitedMembers.length === 1){
+        return this.invitedMembers[0].remoteId;
+      }
+      else
+      {
+        return 0;
+      }
```
Comment (ahamdi) [inline webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue]: Return the full name instead of the username (remoteId)
Try to make it a one line statement

**Review by ahamdi — CHANGES_REQUESTED** [review]
You need to get the inivitation sender from th elistener that creates the notification and then send it to the notification provider

File: component/notification/src/main/java/org/exoplatform/social/notification/channel/template/WebTemplateProvider.java (hunk 1 above)
Comment (nidhalnidhal) [reply component/notification/src/main/java/org/exoplatform/social/notification/channel/template/WebTemplateProvider.java]: ok , i'll remove it

File: component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java (hunk 2 above)
Comment (nidhalnidhal) [reply component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java]: yes

File: component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java (hunk 4 above)
Comment (nidhalnidhal) [reply component/notification/src/main/java/org/exoplatform/social/notification/impl/SpaceNotificationImpl.java]: actually , in our case the editor is the authenticaed user , we are setting the authenticated user as editor in the rest service . So the connected manager is set as editor and will send the invitation. this the best way to do it without modifying in so many other classes .     
https://github.com/Meeds-io/social/blob/develop/component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java#L631

File: component/notification/src/main/java/org/exoplatform/social/notification/plugin/SpaceInvitationPlugin.java (hunk 5 above)
Comment (nidhalnidhal) [reply component/notification/src/main/java/org/exoplatform/social/notification/plugin/SpaceInvitationPlugin.java]: okay

File: extension/notification/src/main/webapp/WEB-INF/notification/templates/SpaceInvitationPlugin.gtmpl (hunk 6 above)
Comment (nidhalnidhal) [reply extension/notification/src/main/webapp/WEB-INF/notification/templates/SpaceInvitationPlugin.gtmpl]: no, actually

File: webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue (hunk 7 above)
Comment (nidhalnidhal) [reply webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue]: yes ok

File: webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue (hunk 8 above)
Comment (nidhalnidhal) [reply webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue]: ok, noted

File: extension/notification/src/main/webapp/WEB-INF/intranet-notification/templates/SpaceInvitationPlugin.gtmpl [hunk 9]
```diff
@@ -6,16 +6,18 @@
...
+                                                : _ctx.appRes("Notification.intranet.message.SpaceInvitationPlugin",senderName, spaceUrl);
         String iconType = (STATUS == "ACCEPTED")  ? "uiIconTick uiIconLightGray" : "uiIconSocGroup uiIconSocLightGray";
       %>
       <div class="contentSmall" data-link="$SPACE_URL">
         <div class="status"><%=actions%></div>
         <div class="content">
         <% if (STATUS != "ACCEPTED") { %>
           <div class="confirm">
-            <a class="btn btn-mini btn-primary action-item" data-rest="$ACCEPT_SPACE_INVITATION_ACTION_URL" href="javascript:void(0)"><%=_ctx.appRes("Notification.label.Accept")%></a>
+            <a class="btn btn-mini btn-primary action-item" data-rest="$ACCEPT_SPACE_INVITATcION_ACTION_URL" href="javascript:void(0)"><%=_ctx.appRes("Notification.label.Accept")%></a>
```
Comment (ahamdi) [inline extension/notification/src/main/webapp/WEB-INF/intranet-notification/templates/SpaceInvitationPlugin.gtmpl]:
```suggestion
            <a class="btn btn-mini btn-primary action-item" data-rest="$ACCEPT_SPACE_INVITATION_ACTION_URL" href="javascript:void(0)"><%=_ctx.appRes("Notification.label.Accept")%></a>
```

File: webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue [hunk 10]
```diff
@@ -161,6 +163,13 @@ export default {
     },
     defaultAvatar() {
       return `${eXo.env.portal.context}/${eXo.env.portal.rest}/v1/social/users/default-image/avatar`;
+    },
+
+    invitedUser() {
+
+      return this.invitedMembers.length === 1 ? this.invitedMembers[0].userName : null;
```
Comment (ahamdi) [inline webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue]: What if we invite multiple users ?

Comment (nidhalnidhal) [reply webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue]: in the task ,it wasn't clear as  they requested a message for one user , but after i discussed the problem with meriem , she told me to add the message ;" users has been invited to space" in case there is many users , so i will do it .

File: webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue [hunk 11]
```diff
@@ -161,7 +161,7 @@ export default {
     },
     defaultAvatar() {
       return `${eXo.env.portal.context}/${eXo.env.portal.rest}/v1/social/users/default-image/avatar`;
-    }
+    },
```
Comment (aelbenney) [inline webapp/portlet/src/main/webapp/space-members/components/SpaceInvitationDrawer.vue]: Remove this unnecessary change please

File: extension/notification/src/main/webapp/WEB-INF/notification/templates/SpaceInvitationPlugin.gtmpl [hunk 12]
```diff
@@ -32,10 +32,10 @@
...
                                                 <p style="margin:0 0 12px; color: #333333; line-height: 20px; font-size:13px; font-family: HelveticaNeue,Helvetica,Arial,sans-serif;">
                                                     <%
-                                                      String spaceUrl = "<a target=\"_blank\" style=\"color: #2f5e92; text-decoration: none; font-weight: bold; font-size:13px; font-family: HelveticaNeue,Helvetica,Arial,sans-serif;\" href=\""+ SPACE_URL + "\">" + _ctx.escapeHTML(SPACE) + "</a>";
+                                                     String spaceUrl = "<a target=\"_blank\" style=\"color: #2f5e92; text-decoration: none; font-weight: bold; font-size:13px; font-family: HelveticaNeue,Helvetica,Arial,sans-serif;\" href=\""+ SPACE_URL + "\">" + _ctx.escapeHTML(SPACE) + "</a>";
                                                     %>
-                                                    <%=_ctx.appRes("Notification.message.SpaceInvitationPlugin", spaceUrl)%>
-                                                </p>
+                                                       <%=_ctx.appRes("Notification.message.SpaceInvitationPlugin", spaceUrl)%>
```
Comment (aelbenney) [inline extension/notification/src/main/webapp/WEB-INF/notification/templates/SpaceInvitationPlugin.gtmpl]: Remove these white spaces in this file please

File: component/notification/src/main/java/org/exoplatform/social/notification/channel/template/WebTemplateProvider.java [hunk 13]
```diff
@@ -966,6 +971,10 @@ protected MessageInfo makeMessage(NotificationContext ctx) {
...
       return false;
     }
-
   }
 }
+
+
+
+
+
```
Comment (aelbenney) [inline component/notification/src/main/java/org/exoplatform/social/notification/channel/template/WebTemplateProvider.java]: To be removed

File: webapp/portlet/webpack.dev.js [hunk 14]
```diff
@@ -1,20 +0,0 @@
...
-let config = merge(webpackCommonConfig, {
-  output: {
-    path: path.resolve(`${exoServerPath}/webapps/${app}/`),
-    filename: 'js/[name].bundle.js',
-    libraryTarget: 'amd'
-  },
-  devtool: 'inline-source-map'
-});
-
-module.exports = config;
```
Comment (aelbenney) [inline webapp/portlet/webpack.dev.js]: Why you deleted the content of this file?

**Review by aelbenney — CHANGES_REQUESTED** [review]

**Review by aelbenney — APPROVED** [review]

**Review by ahamdi — APPROVED** [review]
